app/jobs/submit_to_app_store.rb

```diff
@@ -14,8 +14,9 @@ def perform(submission:)
     # means waiting until the previous lock is released, at which point any
     # modifications made as part of that transaction will have been committed
     submission.with_lock do
-      submit(submission)
+      new_data = submit(submission)
       notify(submission)
+      update_records(new_data, submission)
       submission.update!(submit_to_app_store_completed: true)
```

Comment:
if  new_data['application_state'] == submission.state is true do we still want `submission.update!(submit_to_app_store_completed: true)` to flag as `true` ?

Reply:
I think so, cos either way we have successfully submitted to the app store.